### DISCUSSION TITLE
Assessment/Survey autofill

### DISCUSSION THREAD

**Discussion by kudini** [arnoudkooi/SN-Utils#539] [Ideas]
Autofill functionality is available on portals but currently limited to sc_cat_item only.

In most cases, the existing code should work on OOB assessment pages, but due to this condition:
`if (location.search.includes("sc_cat_item") && location.search.includes("id="))`
it's not being activated, and users aren't informed why it's not working (with /rnd commands remaining in the window).

By adding two small code tweaks (example below):
```
let hasSPAssessmentLayout = document.querySelector('sp-assessment-layout') ? true:false;
if ((location.search.includes("sc_cat_item")||hasSPAssessmentLayout ) && location.search.includes("id="))
```
we can enable the current autofill functionality to be used when testing assessments.

This simple update can significantly improve the user experience during testing.
![obraz](https://github.com/user-attachments/assets/d168273e-dfd3-407e-bd08-8c5e3779ab40)

**Comment by arnoudkooi** [comment]
Good idea, I havent tested, but will add this in next update.
